### PR TITLE
Update CLDR sourced data from 39.0 to 43.1

### PR DESCRIPTION
**Pull request by tpaksu**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This PR updates the `locale-info.php` and `currency-info.php` files initially compiled from [Unicode CLDR](https://cldr.unicode.org/) version 39.0 with the data from version 43.1. The tool to compile the data is located on https://github.com/tpaksu/cldr, and the below diff is generated for the updates (does not contain the locale modifications).

Also, for the multi currency feature on WooPayments, we need the negative amount display formats for each currency available on the store (https://github.com/Automattic/woocommerce-payments/issues/6700), and I've added this specifications to the corresponding files, as WooPayments use the `i18n` files as the source for the multi currency formatting. This change won't affect the formatting of WooCommerce.

There are three methods of displaying the negative amount: 

- `-`: The usual minus sign before all, is being used in 237 countries (except the below list).
- `o-`: The minus sign coming after the symbol, is being used by 7 countries.
    
    ```
    Central African CFA Franc (Central African Republic)
    Swiss Franc (Switzerland)
    Chilean Peso (Chile)
    US Dollar (Ecuador)
    Danish Krone (Greenland)
    Laotian Kip (Laos)
    Venezuelan Bolívar (Venezuela)
    ```
- `o -`: The minus sign coming after the space between the symbol and the amount, is also being used by 7 countries.

    ```
    Aruban Florin (Aruba)
    Euro (Belgium)
    US Dollar (Bonaire, Sint Eustatius and Saba)
    Euro (Netherlands)
    Norwegian Krone (Norway)
    Norwegian Krone (Svalbard and Jan Mayen)
    Surinamese Dollar (Surinam)
    ```

Which are also added to `locale-info.php` file.

#### New Countries

* Added `SHP` currency specifications for `AC` (Ascension Island) country.
* Added `GBP` currency specifications for `TA` (Tristan da Cunha) country.
* Added `EUR` currency specifications for `TF` (French Southern Territories) country.

#### AS changes:
* Changed `name` from `United States (US) dollar` to `US Dollar` for USD

#### BA changes:
* Changed `name` from `Bosnia and Herzegovina convertible mark` to `Bosnia-Herzegovina Convertible Mark` for BAM

#### BM changes:
* Changed `name` from `Bermudian dollar` to `Bermudan Dollar` for BMD

#### BQ changes:
* Changed `name` from `United States (US) dollar` to `US Dollar` for USD

#### BW changes:
* Changed `name` from `Botswana pula` to `Botswanan Pula` for BWP

#### CD changes:
* Changed `currency_pos` from `left_space` to `right_space` for CDF (Congolese Franc)
* Changed `thousand_sep` from `.` to ` ` for CDF (Congolese Franc)
* Changed `default_locale` from `sw_CD` to `fr_CD` for CDF (Congolese Franc)

#### CF changes:
* Changed `currency_pos` from `right_space` to `left` for XAF (Central African CFA Franc)
* Changed `thousand_sep` from ` ` to `.` for XAF (Central African CFA Franc)
* Changed `default_locale` from `fr_CF` to `sg_CF` for XAF (Central African CFA Franc)

#### CU changes:
* Changed `currency_code` from `CUC` to `CUP` for CUP (Cuban Peso)
* Changed `name` from `Cuban convertible peso` to `Cuban Peso` for CUP
* Changed `singular` from `Cuban convertible peso` to `Cuban peso` for CUP (Cuban Peso)
* Changed `plural` from `Cuban convertible pesos` to `Cuban pesos` for CUP (Cuban Peso)

#### CW changes:
* Changed `thousand_sep` from `.` to `,` for ANG (Netherlands Antillean Guilder)
* Changed `decimal_sep` from `,` to `.` for ANG (Netherlands Antillean Guilder)
* Changed `default_locale` from `nl_CW` to `pap_CW` for ANG (Netherlands Antillean Guilder)

#### CZ changes:
* Changed `num_decimals` from `0` to `2` for CZK (Czech Koruna)

#### DG changes:
* Changed `name` from `United States (US) dollar` to `US Dollar` for USD

#### DZ changes:
* Changed `currency_pos` from `left_space` to `right_space` for DZD (Algerian Dinar)

#### EC changes:
* Changed `name` from `United States (US) dollar` to `US Dollar` for USD

#### EH changes:
* Changed `currency_pos` from `left_space` to `right_space` for MAD (Moroccan Dirham)

#### FM changes:
* Changed `name` from `United States (US) dollar` to `US Dollar` for USD

#### GB changes:
* Changed `name` from `Pound sterling` to `British Pound` for GBP

#### GG changes:
* Changed `name` from `Pound sterling` to `British Pound` for GBP

#### GH changes:
* Changed `default_locale` from `ak_GH` to `en_GH` for GHS (Ghanaian Cedi)
* Changed `name` from `Ghana cedi` to `Ghanaian Cedi` for GHS

#### GU changes:
* Changed `name` from `United States (US) dollar` to `US Dollar` for USD

#### GY changes:
* Changed `name` from `Guyanese dollar` to `Guyanaese Dollar` for GYD

#### HR changes:
* Changed `currency_code` from `HRK` to `EUR` for EUR (Euro)
* Changed `name` from `Croatian kuna` to `Euro` for EUR
* Changed `singular` from `Croatian kuna` to `euro` for EUR (Euro)
* Changed `plural` from `Croatian kunas` to `euros` for EUR (Euro)
* Changed `short_symbol` from `kn` to `€` for EUR (Euro)

#### HT changes:
* Changed `currency_code` from `USD` to `HTG` for HTG (Haitian Gourde)
* Changed `name` from `United States (US) dollar` to `Haitian Gourde` for HTG
* Changed `singular` from `US dollar` to `Haitian gourde` for HTG (Haitian Gourde)
* Changed `plural` from `US dollars` to `Haitian gourdes` for HTG (Haitian Gourde)
* Changed `short_symbol` from `$` to `` for HTG (Haitian Gourde)

#### IM changes:
* Changed `name` from `Pound sterling` to `British Pound` for GBP

#### IO changes:
* Changed `name` from `United States (US) dollar` to `US Dollar` for USD

#### JE changes:
* Changed `name` from `Pound sterling` to `British Pound` for GBP

#### KG changes:
* Changed `name` from `Kyrgyzstani som` to `Kyrgystani Som` for KGS

#### LA changes:
* Changed `name` from `Lao kip` to `Laotian Kip` for LAK

#### LS changes:
* Changed `thousand_sep` from `,` to ` ` for LSL (Lesotho Loti)
* Changed `decimal_sep` from `.` to `,` for LSL (Lesotho Loti)
* Changed `default_locale` from `en_LS` to `st_LS` for LSL (Lesotho Loti)

#### LY changes:
* Changed `currency_pos` from `left_space` to `right_space` for LYD (Libyan Dinar)

#### MA changes:
* Changed `currency_pos` from `left_space` to `right_space` for MAD (Moroccan Dirham)

#### MH changes:
* Changed `name` from `United States (US) dollar` to `US Dollar` for USD

#### MM changes:
* Changed `name` from `Burmese kyat` to `Myanmar Kyat` for MMK

#### MN changes:
* Changed `name` from `Mongolian tögrög` to `Mongolian Tugrik` for MNT

#### MP changes:
* Changed `name` from `United States (US) dollar` to `US Dollar` for USD

#### MV changes:
* Changed `thousand_sep` from `.` to `,` for MVR (Maldivian Rufiyaa)
* Changed `decimal_sep` from `,` to `.` for MVR (Maldivian Rufiyaa)
* Changed `default_locale` from `` to `dv_MV` for MVR (Maldivian Rufiyaa)

#### MW changes:
* Changed `currency_pos` from `left` to `left_space` for MWK (Malawian Kwacha)
* Changed `default_locale` from `en_MW` to `ny_MW` for MWK (Malawian Kwacha)

#### NO changes:
* Changed `num_decimals` from `0` to `2` for NOK (Norwegian Krone)

#### PA changes:
* Changed `currency_code` from `USD` to `PAB` for PAB (Panamanian Balboa)
* Changed `name` from `United States (US) dollar` to `Panamanian Balboa` for PAB
* Changed `singular` from `US dollar` to `Panamanian balboa` for PAB (Panamanian Balboa)
* Changed `plural` from `US dollars` to `Panamanian balboas` for PAB (Panamanian Balboa)
* Changed `short_symbol` from `$` to `` for PAB (Panamanian Balboa)

#### PE changes:
* Changed `name` from `Sol` to `Peruvian Sol` for PEN

#### PG changes:
* Changed `currency_pos` from `left` to `right_space` for PGK (Papua New Guinean Kina)
* Changed `default_locale` from `en_PG` to `tpi_PG` for PGK (Papua New Guinean Kina)

#### PH changes:
* Changed `singular` from `Philippine piso` to `Philippine peso` for PHP (Philippine Peso)
* Changed `plural` from `Philippine pisos` to `Philippine pesos` for PHP (Philippine Peso)

#### PK changes:
* Changed `currency_pos` from `left_space` to `left` for PKR (Pakistani Rupee)

#### PL changes:
* Changed `name` from `Polish złoty` to `Polish Zloty` for PLN

#### PR changes:
* Changed `name` from `United States (US) dollar` to `US Dollar` for USD

#### PS changes:
* Changed `currency_code` from `JOD` to `ILS` for ILS (Israeli New Shekel)
* Changed `num_decimals` from `3` to `2` for ILS (Israeli New Shekel)
* Changed `name` from `Jordanian dinar` to `Israeli New Shekel` for ILS
* Changed `singular` from `Jordanian dinar` to `Israeli new shekel` for ILS (Israeli New Shekel)
* Changed `plural` from `Jordanian dinars` to `Israeli new shekels` for ILS (Israeli New Shekel)

#### PW changes:
* Changed `name` from `United States (US) dollar` to `US Dollar` for USD

#### PY changes:
* Changed `thousand_sep` from `.` to `,` for PYG (Paraguayan Guarani)
* Changed `decimal_sep` from `,` to `.` for PYG (Paraguayan Guarani)
* Changed `default_locale` from `es_PY` to `gn_PY` for PYG (Paraguayan Guarani)
* Changed `name` from `Paraguayan guaraní` to `Paraguayan Guarani` for PYG

#### QA changes:
* Changed `singular` from `Qatari rial` to `Qatari riyal` for QAR (Qatari Riyal)
* Changed `plural` from `Qatari rials` to `Qatari riyals` for QAR (Qatari Riyal)

#### SA changes:
* Changed `thousand_sep` from `,` to `.` for SAR (Saudi Riyal)
* Changed `decimal_sep` from `.` to `,` for SAR (Saudi Riyal)

#### SE changes:
* Changed `num_decimals` from `0` to `2` for SEK (Swedish Krona)

#### SH changes:
* Changed `name` from `Saint Helena pound` to `St. Helena Pound` for SHP

#### SJ changes:
* Changed `num_decimals` from `0` to `2` for NOK (Norwegian Krone)

#### SL changes:
* Changed `currency_code` from `SLL` to `SLE` for SLE (Sierra Leonean Leone)
* Changed `num_decimals` from `0` to `2` for SLE (Sierra Leonean Leone)

#### ST changes:
* Changed `name` from `São Tomé and Príncipe dobra` to `São Tomé & Príncipe Dobra` for STN

#### SV changes:
* Changed `name` from `United States (US) dollar` to `US Dollar` for USD

#### TC changes:
* Changed `name` from `United States (US) dollar` to `US Dollar` for USD

#### TL changes:
* Changed `name` from `United States (US) dollar` to `US Dollar` for USD

#### TM changes:
* Changed `name` from `Turkmenistan manat` to `Turkmenistani Manat` for TMT

#### TN changes:
* Changed `currency_pos` from `left_space` to `right_space` for TND (Tunisian Dinar)

#### TT changes:
* Changed `name` from `Trinidad and Tobago dollar` to `Trinidad & Tobago Dollar` for TTD

#### TW changes:
* Changed `default_locale` from `zh_Hant` to `zh_Hant_TW` for TWD (New Taiwan Dollar)

#### UM changes:
* Changed `name` from `United States (US) dollar` to `US Dollar` for USD

#### US changes:
* Changed `name` from `United States (US) dollar` to `US Dollar` for USD

#### UZ changes:
* Changed `default_locale` from `uz_AF` to `uz_UZ` for UZS (Uzbekistani Som)

#### VE changes:
* Changed `name` from `Bolívar soberano` to `Venezuelan Bolívar` for VES

#### VG changes:
* Changed `name` from `United States (US) dollar` to `US Dollar` for USD

#### VI changes:
* Changed `name` from `United States (US) dollar` to `US Dollar` for USD

#### VN changes:
* Changed `name` from `Vietnamese đồng` to `Vietnamese Dong` for VND

#### WS changes:
* Changed `name` from `Samoan tālā` to `Samoan Tala` for WST

#### ZA changes:
* Changed `thousand_sep` from ` ` to `,` for ZAR (South African Rand)
* Changed `decimal_sep` from `,` to `.` for ZAR (South African Rand)

#### ZW changes:
* Changed `name` from `United States (US) dollar` to `US Dollar` for USD

### How to test the changes in this Pull Request:

The Data API tests and E2E tests on this PR should pass.

**Manual Test:**
1. While onboarding, select `French Southern Territories` as your country, and finish onboarding. If you have already finished onboarding, you can go to **WooCommerce > Orders**, then on the above tab drawer, choose **Help > Setup Wizard**, then click on the **Setup Wizard** button to re-access the onboarding workflow.
2. After you finish onboarding, go to `WooCommerce > Settings > General` and on the Currency select, you should see `Euro (€)` as the selected currency for that store.

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [x] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [x] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 
Update CLDR sourced locale data from CLDR 39.0 to 43.1

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/tests/api-core-tests/tests/data/data-crud.test.js
Comment (lsinger) [inline plugins/woocommerce/tests/api-core-tests/tests/data/data-crud.test.js:1530]: I'm not an expert on the politics that might lie behind this, but this seems ... notable? Do we have somehow who can explain the background here before merging?

File: plugins/woocommerce/tests/api-core-tests/tests/data/data-crud.test.js
Comment (lsinger) [inline plugins/woocommerce/tests/api-core-tests/tests/data/data-crud.test.js]: This change might not be intended to be merged?

File: plugins/woocommerce/i18n/currency-info.php
Comment (lsinger) [inline plugins/woocommerce/i18n/currency-info.php]: Looking at the release calendar, his should probably be:

```suggestion
 * @version 8.2.0
```

File: plugins/woocommerce/i18n/locale-info.php
Comment (lsinger) [inline plugins/woocommerce/i18n/locale-info.php]: See my comment on `plugins/woocommerce/i18n/currency-info.php` -- this should probably be:

```suggestion
 * @version 8.2.0
```

File: plugins/woocommerce/i18n/locale-info.php
Comment (lsinger) [inline plugins/woocommerce/i18n/locale-info.php:2920]: See my comment on `plugins/woocommerce/tests/api-core-tests/tests/data/data-crud.test.js` -- even if this is just a normal non-political, technical fix for something, this seems like a thing we should make sure we all understand well before merging.

**Review by lsinger — CHANGES_REQUESTED** [review]
I left a few notes and questions regarding some specific changes. 

I was unable to verify the manual test. When I chose `French Southern Territories` in the onboarding, that was set correctly in Settings -- but currency was set to `United States (US) $`. This is a local and completely fresh wp-env install just on this branch. Do I need to run some sort of build step before this works, maybe?

File: plugins/woocommerce/tests/api-core-tests/tests/data/data-crud.test.js
Comment (tpaksu) [reply plugins/woocommerce/tests/api-core-tests/tests/data/data-crud.test.js:1530]: Thanks for asking this question, this was also confusing for me too. 

I searched it online, and read a few websites telling about the currencies used in Palestine, and it seems Palestine doesn't have it's own currency, and there are three major currencies used, the US Dollar, the Israeli Shekel, and the Jordanian Dinar for trade, which the websites tell in common. But some websites talk about a legal tender used in West Bank and the Gaza Strip, and it changed from JOD to ILS after the Paris Protocol in 1994. 

https://www.monito.com/en/what-is-the-currency-in/palestine
https://israelrail.com/israeli-currency-new-israeli-shekel/
https://gopalestine.org/about-palestine__trashed/currency-and-money-in-palestine/
https://www.worldtravelguide.net/guides/middle-east/palestinian-national-authority/money-duty-free/
https://qr.ae/pyLUIP
https://globalfinancialdata.com/palestine

But they do not actually tell whether Palestine uses a definitive currency. They mostly talk about Israeli occupied territories using ILS as a legal tender.

And according to CLDR, there are currently two currencies used (ILS and JOD): 

https://github.com/tpaksu/cldr/blob/main/cldr/cldr-core/supplemental/currencyData.json#L2617-L2640

Besides those, I don't know exactly what's true. Maybe we can ask some A11ns who are living in that area?

File: plugins/woocommerce/tests/api-core-tests/tests/data/data-crud.test.js
Comment (tpaksu) [reply plugins/woocommerce/tests/api-core-tests/tests/data/data-crud.test.js]: Yes, that was for fixing the API tests on local. A test was failing unless I set that to `false` explicitly. I also couldn't pass the permission checks no matter if I used the keys, or the credentials. The keys make the `reset` stage pass, but the tests to give 401's, and the credentials didn't work for the resetting step. Maybe I missed something.

File: plugins/woocommerce/i18n/currency-info.php
Comment (tpaksu) [reply plugins/woocommerce/i18n/currency-info.php]: Thank you! I'll update both files.

File: plugins/woocommerce/i18n/locale-info.php
Comment (tpaksu) [reply plugins/woocommerce/i18n/locale-info.php:2920]: Definitely I agree. I'm not sure how to convert this to accept multiple currencies for some countries, as some other countries use 2-3 currencies actively too. Thanks for bringing this up!

**Comment by tpaksu** [comment]
@lsinger, about the testing instructions, I have to make a confession, I copied it from https://github.com/woocommerce/woocommerce/pull/30216 on the same subject from 2 years ago, since those files (I believe) are not used except the currency selection according to the country that user selected on onboarding. 

Other than that, I have no idea. I asked it on p1694427870806519-slack-C0E1AV8T0 and they suggested me to create a patch that mimics the change on https://github.com/Automattic/woocommerce-payments/pull/7120 on the Orders screen for the testing instructions. 

Since `wc_price` doesn't use `locale-info.php`, the patch will be rewriting the whole WooPayments code in Core, so I used the old testing instructions. And I've also tested, and it seems the re-onboarding test doesn't work anymore. It doesn't change the currency after the initial onboarding. 

Maybe deleting the option holding the WooCommerce currency will make it work, but I'm not comfortable enough to say that's the way to test it.

File: plugins/woocommerce/i18n/locale-info.php
Comment (tpaksu) [reply plugins/woocommerce/i18n/locale-info.php]: Addressed in https://github.com/woocommerce/woocommerce/pull/40101/commits/3de8ed8c55df591024db9f25148e908d87376228.

File: plugins/woocommerce/i18n/currency-info.php
Comment (tpaksu) [reply plugins/woocommerce/i18n/currency-info.php]: Addressed in https://github.com/woocommerce/woocommerce/pull/40101/commits/3de8ed8c55df591024db9f25148e908d87376228.

File: plugins/woocommerce/tests/api-core-tests/tests/data/data-crud.test.js
Comment (tpaksu) [reply plugins/woocommerce/tests/api-core-tests/tests/data/data-crud.test.js]: Addressed in https://github.com/woocommerce/woocommerce/pull/40101/commits/3de8ed8c55df591024db9f25148e908d87376228.

File: plugins/woocommerce/i18n/locale-info.php
Comment (tpaksu) [reply plugins/woocommerce/i18n/locale-info.php:2920]: According to CLDR, these countries have more than one currencies currently active: 


- **BT (Bhutan)   :** INR, [BTN](https://en.wikipedia.org/wiki/Bhutanese_ngultrum) (Pegged to INR 1:1) 
- **CU (Cuba)     :** [CUP](https://en.wikipedia.org/wiki/Cuban_peso), [CUC](https://en.wikipedia.org/wiki/Cuban_convertible_peso) CUP active since 2020
- **HT (Haiti)    :** [HTG](https://en.wikipedia.org/wiki/Haitian_gourde), the active one according to https://en.wikipedia.org/wiki/Haiti, USD 
- **LS (Lesotho)  :** ZAR, [LSL](https://en.wikipedia.org/wiki/Lesotho_loti)
- **NA (Namibia)  :** ZAR, [NAD](https://en.wikipedia.org/wiki/Namibian_dollar)
- **PA (Panama)   :** [PAB](https://en.wikipedia.org/wiki/Panamanian_balboa), USD
- **PS (Palestine):** ILS, JOD (?)

Comment (tpaksu) [reply plugins/woocommerce/i18n/locale-info.php:2920]: I could confirm the currencies for: 

- Cuba: https://www.gov.uk/foreign-travel-advice/cuba/money
- Haiti: https://www.gov.uk/foreign-travel-advice/haiti/money
- Lesotho: https://www.gov.uk/foreign-travel-advice/lesotho/money

But Bhutan, Namibia, Panama and Palestine still is unknown.

**Comment by tpaksu** [comment]
@lsinger I found why the testing instructions are not working. The onboarding setup has changed to a React based one from the PHP based one, and I guess the React based one doesn't include that "update WooCommerce currency by selected country" thing. So it makes the changes untestable, as the `locale-info.php` seems to be only used by the REST continents query, which is tested by the playwright API tests, which I've updated, and this part of the onboarding, which I couldn't find where the `updateAndPersistSettingsForGroup` method effects (yet): 

https://github.com/woocommerce/woocommerce/blob/3de8ed8c55df591024db9f25148e908d87376228/plugins/woocommerce-admin/client/profile-wizard/steps/store-details/index.js#L126-L164

**Comment by tpaksu** [comment]
An issue is created to solve the onboarding problem which is not changing the currency based on the onboarding country: https://github.com/woocommerce/woocommerce/issues/40135

File: plugins/woocommerce/tests/api-core-tests/tests/data/data-crud.test.js
Comment (lsinger) [reply plugins/woocommerce/tests/api-core-tests/tests/data/data-crud.test.js:1530]: I discussed this with a few colleagues and thought about it more. In, for example, `i18n/countries.php`, we explicitly call out following the Unicode CLDR recommendation. They're the experts on these issues and provide the recommendations that we've decided to follow. Let's go forward with this one.

File: plugins/woocommerce/i18n/locale-info.php
Comment (lsinger) [reply plugins/woocommerce/i18n/locale-info.php:2920]: This probably warrants the same treatment as my comment on `plugins/woocommerce/tests/api-core-tests/tests/data/data-crud.test.js` -- let's trust and go with the Unicode recommendation here.

**Review by lsinger — APPROVED** [review]
My comments have been addressed by 3de8ed8c55df591024db9f25148e908d87376228 or I've discussed and resolved internally. 

Since the changes cannot be manually tested as you discussed in the comments, I'd recommend Solaris to give this a review before merging (possibly in addition to the `needs analysis` tag).

File: plugins/woocommerce/tests/api-core-tests/tests/data/data-crud.test.js
Comment (tpaksu) [reply plugins/woocommerce/tests/api-core-tests/tests/data/data-crud.test.js:1530]: Yes, but the CLDR doesn't explicitly call out any currencies on the four countries I've mentioned in the last linked review comment of yours. Maybe we should develop a feature that allows selecting the right currency for those countries which have inseperable currencies on onboarding?

**Comment by alopezari** [comment]
Hi @tpaksu!

Thank you for the hard work on this PR.

This is a big change related to i18n, which should be considered high-impact due to the high number of users that it could harm in case something goes wrong, plus the history of i18n issues we've seen recently and the lack of proper i18n automated testing we have yet.

I think we should find a way to test this PR properly in order to minimize the risks if possible. Is there any way we could test it? Even if we need to add additional code snippets or something, that would be fine. Thanks!

**Comment by tpaksu** [comment]
Hi @alopezari! 

> This is a big change related to i18n, which should be considered high-impact due to the high number of users that it could harm in case something goes wrong, plus the history of i18n issues we've seen recently and the lack of proper i18n automated testing we have yet.

Sure, I only thought that it's still used only for the country picker on the onboarding screen, as I originally modified it to include the locales and formats for each locale 2 years ago, that was the case. But it seems it's drastically improved this time 😬 

> I think we should find a way to test this PR properly in order to minimize the risks if possible. Is there any way we could test it? Even if we need to add additional code snippets or something, that would be fine. Thanks!

Yes, the current testing instruction (which is included in the PR description) is the only thing I know that the `locale-info.php` is being used on WooCommerce Core, and I don't know much about the other usages, directly on indirectly, if you can help me, or find the right people on the Core team that could help me to prepare other testing instructions, I'll really appreciate! The current testing instructions I provided seems to have a bug, and is being fixed in https://github.com/woocommerce/woocommerce/issues/40135 in the meantime.

**Comment by alopezari** [comment]
Thanks for the context @tpaksu!

I've been trying to gather info on how to test this PR and it seems CLDR-related changes might be subject to different kinds of implementations which we might need to discuss in order to establish a common process for addressing it.

We will raise a discussion about this soon and I was wondering if this PR was very urgent or if it can wait until the we get some conclusions from it and the process is settled.

cc @barryhughes

**Comment by tpaksu** [comment]
Thanks for letting me know @alopezari! 

It doesn't have an urgency, but it is linked to https://github.com/Automattic/woocommerce-payments/pull/7120 that's depending on it. If you let me know about where the discussion happens, I'd like to follow it too to learn what these changes affect and be careful about the future changes. 

The fix for the onboarding country affecting the store currency seems to be merged to core, the PR should be testable by the provided testing instruction now.


**Manual Test:**
1. While onboarding, select `French Southern Territories` as your country, and finish onboarding. If you have already finished onboarding, you can go to **WooCommerce > Orders**, then on the above tab drawer, choose **Help > Setup Wizard**, then click on the **Setup Wizard** button to re-access the onboarding workflow.
2. After you finish onboarding, go to `WooCommerce > Settings > General` and on the Currency select, you should see `Euro (€)` as the selected currency for that store.
3. Try it with different countries, and the currencies should match with the onboarded country.

**Comment by lsinger** [comment]
> you should see Euro (€) as the selected currency for that store

@tpaksu heads up that even after pulling this branch anew and starting a whole new store, I'm still seeing `United States (US) dollar $` for the currency after going through onboarding with `French Southern Territories` as the location. 

The `(US) dollar` string is still present in a few files (see screenshot), so I wonder whether more files need to be updated? 🤔 

Screenshot 2023-09-20 at 16 02 06

**Comment by tpaksu** [comment]
> The (US) dollar string is still present in a few files (see screenshot), so I wonder whether more files need to be updated?

I don't think so, those changed files should be enough to work. maybe @moon0326 can assist with this as he worked on fixing this issue in https://github.com/woocommerce/woocommerce/issues/40135? 

The TF country specification is like this

https://github.com/woocommerce/woocommerce/blob/240a86d1d8655cd0225fd7a5588046a1ee71a059/plugins/woocommerce/i18n/locale-info.php#L3686-L3702

**Comment by lsinger** [comment]
> I don't think so, those changed files should be enough to work.

@tpaksu but looking beyond yet, even if the testing instructions did work, I wonder: the fact that we can still find the old string in a few places might tell us that we're not using the CLDR data in all the places we should be using it? 

And going beyond even *that*, I wonder: these are not only changes in display in this PR, but changes that have the potential to, say, change the currency of the store. Are we sure that this can only happen in new stores and not in existing stores? If it can happen in existing stores -- how can we make sure this doesn't disrupt anything?

**Comment by tpaksu** [comment]
@lsinger I have to read all the parts that might eventually be affected from these changes on a codebase which I'm not even familiar with to answer that, and it may take a very long time for me to answer it, than you getting the answer from someone who worked on i18n on WC Core :) If you want me to do the research, I'll dedicate some time for it and get the answer. 

Actually, I wonder the same things that you asked in your questions too, to be ready for the future PR's in this context.

**Comment by lsinger** [comment]
@tpaksu sounds like we both don't have the context to be able to assess this PR -- please feel free to suggest someone whom you think might be a better match. I'll also try to figure out who might be able to help here.

**Comment by tpaksu** [comment]
Closing the issue because the source issue possibly has a smaller fix than updating the CLDR data on WC Core. This would be nice to cover all negative format possibilities for each currency, but it's not needed.

**Pull request closed without merging**
